cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/ExecutionSparkContext.java

```diff
@@ -160,7 +242,8 @@ public ExecutionSparkContext(SparkSpecification specification, Id.Program progra
 
   @Override
   public synchronized <T extends Dataset> T getDataset(String name, Map<String, String> arguments) {
-    Map<String, String> datasetArgs = RuntimeArguments.extractScope(Scope.DATASET, name, getRuntimeArguments());
+    Map<String, String> datasetArgs = new HashMap<>();
```

Comment:
If the contract of the `extractScope` always return a new HashMap, then we don't need to copy it one more time here.